### PR TITLE
app: Use std.ChildProcess instead of POSIX calls

### LINKED ISSUE #739: app: windows target CI failure
https://github.com/hexops/mach/actions/runs/4590437389/jobs/8105981801

### PR DESCRIPTION
**Pull request by tato**
fixes #739

In my testing it builds like in the fork/exec implementation. But I couldn't get any of the examples to run under wasm, even under the old implementation.

- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

File: app/Builder.zig [hunk 1]
```diff
@@ -88,48 +84,42 @@ pub fn run(self: *Builder) !void {
...
+
+    const args = try self.buildArgs(args_arena.allocator());
 
-    return std.os.execvpeZ(
-        argv[0].?,
-        @ptrCast([*:null]const ?[*:0]const u8, argv),
-        @ptrCast([*:null]const ?[*:0]const u8, std.os.environ.ptr),
-    );
+    var child = std.ChildProcess.init(args, allocator);
+    _ = try child.spawnAndWait();
```
Comment (alichraghi) [inline app/Builder.zig:97]: doesn't check exit status

**Comment by alichraghi** [comment]
Hi @tato. thanks for working on this. i just tested it and it works fine on linux.
you can test it using:
(under mach-examples repo)
```
<PATH_TO_MACH_REPO>/zig-out/bin/mach build sysaudio --serve
```

File: app/Builder.zig (hunk 1 above)
Comment (tato) [reply app/Builder.zig:97]: True 😅 I guess it should just exit if the result is == .Signal / .Stopped / .Unknown too?

Comment (alichraghi) [reply app/Builder.zig]: exactly

**Review by alichraghi — APPROVED** [review]

**Comment by alichraghi** [comment]
this doesn't close #739 tho. because there's another fork in https://github.com/hexops/mach/blob/1841ed056d41dd19b0a7d238a52b1279547b6477/app/Builder.zig#L323 which can't be solved easily since `pipe` won't work this way so we can't collect stderr

**Comment by tato** [comment]
Should be solvable with std.ChildProcess too in a cross-platform way. I didn't realise there was another `fork` call 😅

**Comment by alichraghi** [comment]
hmm not sure. needs testing. also `std.ChildProcess` has been renamed to `std.process.Child`

**Comment by alichraghi** [comment]
https://ziglang.org/documentation/master/std/src/child_process.zig.html#L48

**Comment by alichraghi** [comment]
nothing yet. watcher will call it once #622 get fixed. also LGTM (this fixes #739)

**Review by slimsag — APPROVED** [review]

**Pull request merged**